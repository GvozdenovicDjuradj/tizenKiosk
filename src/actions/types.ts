import { Actions as KioskActions } from "./kiosk"
import { Action as ValidationActions } from "./validation"
import { PrintingAction } from "./printing"

export const APPLICATION_START = "APPLICATION_START"
export type APPLICATION_START = typeof APPLICATION_START

const CANCEL = "CANCEL"
const FAILURE = "FAILURE"
const REQUEST = "REQUEST"
const RESET = "RESET"
const SUCCESS = "SUCCESS"

const _ADD_TO_QUEUE = "ADD_TO_QUEUE"
const _ANSWERED = "ANSWERED"
const _CHECK_IN = "CHECK_IN"
const _CUSTOMER = "CUSTOMER"
const _ERROR = "ERROR"
const _EVENT = "EVENT"
const _FIELD_UPDATE = "FIELD_UPDATE"
const _KIOSK = "KIOSK"
const _LOGIN = "LOGIN"
const _LOGOUT = "LOGOUT"
const _POST_ANSWERS = "POST_ANSWERS"
const _QUESTIONS = "QUESTIONS"
const _QUEUE_DATA = "QUEUE_DATA"
const _REGISTER = "REGISTER"
const _SET_BOOKING_REF = "SET_BOOKING_REF"
const _SET_CALLING_CODE = "SET_CALLING_CODE"
const _SET_COUNTRY = "SET_COUNTRY"
const _SET_CURRENT = "SET_CURRENT"
const _SET_EMAIL = "SET_EMAIL"
const _SET_LANGUAGE = "SET_LANGUAGE"
const _SET_MOBILE = "SET_MOBILE"
const _SET_NAME = "SET_NAME"
const _SET_ORDER = "SET_ORDER"
const _SET_NOTES = "SET_NOTES"
const _SET_GROUP_SIZE = "SET_GROUP_SIZE"
const _SET_PRODUCT = "SET_PRODUCT"
const _SETTINGS = "SETTINGS"
const _TOGGLE_OTHER = "TOGGLE_OTHER"
const _LOAD_SCREEN_SAVER_DATA = "LOAD_SCREEN_SAVER_DATA"
const _LOAD_PRIVACY_POLICY_DATA = "LOAD_PRIVACY_POLICY_DATA"
const _ADD_PRIVATE_CUSTOMER_TO_QUEUE = "ADD_PRIVATE_CUSTOMER_TO_QUEUE"

export function createActionTypes(type: string) {
  return {
    [CANCEL]: `${type}_${CANCEL}`,
    [FAILURE]: `${type}_${FAILURE}`,
    [REQUEST]: `${type}_${REQUEST}`,
    [RESET]: `${type}_${RESET}`,
    [SUCCESS]: `${type}_${SUCCESS}`,
  }
}

export const ADD_CUSTOMER_TO_QUEUE_START = 'ADD_CUSTOMER_TO_QUEUE_START';
export const ADD_CUSTOMER_TO_QUEUE_STOP = 'ADD_CUSTOMER_TO_QUEUE_STOP';

export const PRIVACY_POLICY_AGREE = "PRIVACY_POLICY_AGREE"
export const PRIVACY_POLICY_DISAGREE = "PRIVACY_POLICY_DISAGREE"
export const LOGIN = createActionTypes(_LOGIN)
export const LOGOUT = createActionTypes(_LOGOUT)
export const FIELD_UPDATE = createActionTypes(_FIELD_UPDATE)
export const KIOSK = {
  CHECK_IN: {
    ...createActionTypes(`${_KIOSK}_${_CHECK_IN}`),
    SET_CALLING_CODE: `${_KIOSK}_${_CHECK_IN}_${_SET_CALLING_CODE}`,
    SET_COUNTRY: `${_KIOSK}_${_CHECK_IN}_${_SET_COUNTRY}`,
    SET_EMAIL: createActionTypes(`${_KIOSK}_${_CHECK_IN}_${_SET_EMAIL}`),
    SET_MOBILE: createActionTypes(`${_KIOSK}_${_CHECK_IN}_${_SET_MOBILE}`),
    SET_ORDER: createActionTypes(`${_KIOSK}_${_CHECK_IN}_${_SET_ORDER}`),
  },
  CUSTOMER: {
    ADD_TO_QUEUE: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_ADD_TO_QUEUE}`),
    SET_CALLING_CODE: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_CALLING_CODE}`),
    SET_COUNTRY: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_COUNTRY}`),
    SET_EMAIL: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_EMAIL}`),
    SET_MOBILE: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_MOBILE}`),
    SET_NAME: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_NAME}`),
    SET_GROUP_SIZE: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_GROUP_SIZE}`),
    SET_ORDER: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_ORDER}`),
    SET_NOTES: createActionTypes(`${_KIOSK}_${_CUSTOMER}_${_SET_NOTES}`),
  },
  EVENT_CHECK_IN: {
    ...createActionTypes(`${_KIOSK}_${_EVENT}_${_CHECK_IN}`),
    SET_EMAIL: createActionTypes(`${_KIOSK}_${_EVENT}_${_CHECK_IN}_${_SET_EMAIL}`),
    SET_BOOKING_REF: createActionTypes(`${_KIOSK}_${_EVENT}_${_CHECK_IN}_${_SET_BOOKING_REF}`),
  },
  FIELD_UPDATE: createActionTypes(`${_KIOSK}_${_FIELD_UPDATE}`),
  QUEUE_DATA: createActionTypes(`${_KIOSK}_${_QUEUE_DATA}`),
  REGISTER: createActionTypes(`${_KIOSK}_${_REGISTER}`),
  SET_COUNTRY: createActionTypes(`${_KIOSK}_${_SET_COUNTRY}`),
  SET_LANGUAGE: createActionTypes(`${_KIOSK}_${_SET_LANGUAGE}`),
  SET_PRODUCT: createActionTypes(`${_KIOSK}_${_SET_PRODUCT}`),
  SET_SUBPRODUCT: createActionTypes(`${_KIOSK}_SET_SUBPRODUCT`),
  SETTINGS: createActionTypes(`${_KIOSK}_${_SETTINGS}`),
  LOAD_SCREEN_SAVER_DATA: createActionTypes(`${_KIOSK}_${_LOAD_SCREEN_SAVER_DATA}`),
  LOAD_PRIVACY_POLICY_DATA: createActionTypes(`${_KIOSK}_${_LOAD_PRIVACY_POLICY_DATA}`),
  ADD_PRIVATE_CUSTOMER_TO_QUEUE: createActionTypes(`${_KIOSK}_${_ADD_PRIVATE_CUSTOMER_TO_QUEUE}`)
}

export const VALIDATION_STATE_CHANGE = "VALIDATION_STATE_CHANGE"

export const OTHER_ACTION = "OTHER_ACTION"

export const APP = {
  CHANGE_INITIAL_PAGE: createActionTypes("CHANGE_INITIAL_PAGE"),
  KEYBOARD_STATE_CHANGE: "APP_KEYBOARD_STATE_CHANGE",
  PHONE_STATE_PERMISSION: createActionTypes("PHONE_STATE_PERMISSION"),
  NAVIGATE: createActionTypes("APP_NAVIGATE"),
  OFFLINE: createActionTypes("APP_OFFLINE"),
  ONLINE: createActionTypes("APP_ONLINE"),
  SECRET_TAP: createActionTypes("APP_SECRET_TAP"),
  SET_IDLE_INTERVAL: createActionTypes("APP_SET_IDLE_INTERVAL"),
  SHUTDOWN: createActionTypes("APP_SHUTDOWN"),
  RESET: {
    FAILURE: `APP_${RESET}_${FAILURE}`,
    REQUEST: `APP_${RESET}_${REQUEST}`,
    SUCCESS: `APP_${RESET}_${SUCCESS}`,
  },
  START: createActionTypes("APP_START"),
  GO_INITIAL_SCREEN: createActionTypes("APP_GO_INITIAL_SCREEN")
}

export const MODAL = {
  SHOW: createActionTypes("MODAL_SHOW"),
  HIDE: createActionTypes("MODAL_HIDE")
}

export const ERROR_SHOW = "ERROR_SHOW"
export type ERROR_SHOW = typeof ERROR_SHOW

export const ERROR_RESET = `${_ERROR}_${RESET}`
export type ERROR_RESET = typeof ERROR_RESET

export const KIOSK_SETTINGS_FORM = {
  CHECK_NETWORK: createActionTypes("KIOSK_SETTINS_FORM_CHECK_NETWORK"),
  FIELD_UPDATE: createActionTypes("KIOSK_SETTINS_FORM_FIELD_UPDATE"),
  SET_INITIAL_VALUES: createActionTypes("KIOSK_SETTINS_FORM_SET_INITIAL_VALUES"),
  SUBMIT: createActionTypes("KIOSK_SETTINS_FORM_SUBMIT"),
  VALIDATE: createActionTypes("KIOSK_SETTINGS_FORM_VALIDATE"),
}

export const QUESTIONS = {
  ...createActionTypes(`${_QUESTIONS}`),
  ANSWERED: `${_QUESTIONS}_${_ANSWERED}`,
  ANSWERS: { RESET: `${_QUESTIONS}_ANSWERS_${RESET}` },
  NEXT: `${_QUESTIONS}_NEXT`,
  POST_ANSWERS: createActionTypes(`${_QUESTIONS}_${_POST_ANSWERS}`),
  RESET: `${_QUESTIONS}_${RESET}`,
  SET_CURRENT: `${_QUESTIONS}_${_SET_CURRENT}`,
  TOGGLE_OTHER: `${_QUESTIONS}_${_TOGGLE_OTHER}`,
}

export const SCREEN_SAVER = {
  ...createActionTypes("SCREEN_SAVER")
}

export type Action = KioskActions | ValidationActions | PrintingAction
