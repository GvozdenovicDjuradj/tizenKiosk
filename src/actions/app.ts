import { Action, APP, APPLICATION_START } from "./types"
import { AppScreens } from "../interfaces"

export const applicationStart = (): Action => ({ type: APPLICATION_START })

export const initApp =  (): Action => ({type: APP.START.REQUEST})

export const stopApp = (): Action => ({ type: APP.SHUTDOWN.REQUEST })

export const changeInitialPage = (payload: AppScreens): Action => ({
  type: APP.CHANGE_INITIAL_PAGE.REQUEST,
  payload,
})

export const getPermissionReadPhoneState = () => ({
  type: APP.PHONE_STATE_PERMISSION.REQUEST
})

export const deactivateApp = (): Action => ({ type: APP.RESET.REQUEST })
