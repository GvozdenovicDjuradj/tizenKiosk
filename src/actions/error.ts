import { AnyAction } from "redux"
import { ERROR_SHOW, ERROR_RESET } from "./types"

export interface ShowError extends AnyAction {
  type: ERROR_SHOW;
  payload: string;
}

export interface ResetError extends AnyAction {
  type: ERROR_RESET
}

export type ErrorAction = ResetError | ShowError

export const showError = (payload: string): ShowError => ({
  type: ERROR_SHOW,
  payload
})

export const resetError = (): ResetError => ({ type: ERROR_RESET })
