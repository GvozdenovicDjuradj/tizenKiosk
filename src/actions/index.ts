export * from "./app"
export * from "./error"
export * from "./kiosk"
export * from "./kioskSettings"
export * from "./printing"
export * from "./navigationActions"
export * from "./screenSaver"
