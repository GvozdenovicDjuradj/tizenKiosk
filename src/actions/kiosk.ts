import { Action, AnyAction } from "redux"
import { APP, KIOSK, PRIVACY_POLICY_AGREE, PRIVACY_POLICY_DISAGREE } from "./types"
import {
  AddCustomerToQueueReset,
  AddCustomerToQueueSuccess,
  AddPrivateCustomerToQueueRequest,
  AddPrivateCustomerToQueueSuccess,
  AnyProduct,
  CustomerInQueue,
  DatePayload,
  ErrorPayload,
  FieldUpdatePayload,
  GetKioskDataSuccess,
  GetKioskSettingsSuccess,
  KioskSettings,
  Language,
  LanguagePayload,
  LoadPrivacyPolicySuccess,
  PrivacyPolicy,
  Product,
  ProductPayload,
  QuestionAnswerPayload,
  RegisterKioskFieldUpdate,
  RegisterKioskSuccess,
  StringPayload,
  NumberPayload
} from "../interfaces"
import { GET_KIOSK_ID_AND_DOMAIN_FROM_URL, SET_KIOSK_ID, SET_KIOSK_SERIAL, SET_KIOSK_URL } from "../constants/kiosk"

export const registerKiosk = (showPrint: boolean): AnyAction =>
  ({ type: KIOSK.REGISTER.REQUEST, hasPrinter: showPrint })

export const registerKioskFieldUpdate = (payload: FieldUpdatePayload): RegisterKioskFieldUpdate => ({
  payload,
  type: KIOSK.FIELD_UPDATE.REQUEST,
})

export const loadKioskSettings = (): Action => ({ type: KIOSK.SETTINGS.REQUEST })

export const loadScreenSaverData = (): Action => ({ type: KIOSK.LOAD_SCREEN_SAVER_DATA.REQUEST })

export const loadKioskQueueData = (): Action => ({ type: KIOSK.QUEUE_DATA.REQUEST })

interface KioskSettingsSuccessAction {
  type: string,
  payload: KioskSettings
}

export const kioskSettingsSuccess = (kioskSettings: KioskSettings): KioskSettingsSuccessAction => ({
  payload: kioskSettings,
  type: KIOSK.SETTINGS.SUCCESS
})

export const changeLanguage = (payload: Language): LanguagePayload => ({
  payload,
  type: KIOSK.SET_LANGUAGE.REQUEST,
})

export const setVenueCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.SET_COUNTRY.REQUEST
})

export const setGroupSize = (payload: number): NumberPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_GROUP_SIZE.REQUEST,
})

export const setCustomerName = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NAME.REQUEST,
})

export const setCustomerEmail = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_EMAIL.REQUEST,
})

export const setCustomerCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_COUNTRY.REQUEST,
})

export const setCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_CALLING_CODE.REQUEST,
})

export const setCustomerMobile = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_MOBILE.REQUEST,
})

export const setCustomerOrder = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_ORDER.REQUEST,
})

export const setNotes = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NOTES.REQUEST,
})

export const setProduct = (payload: Product | AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_PRODUCT.REQUEST,
})

export const setSubProduct = (payload: AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_SUBPRODUCT.REQUEST,
})

export const addCustomerToQueue = (): Action => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.REQUEST,
})

export const setAddCustomerToQueueError = (payload: string): StringPayload => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.FAILURE,
  payload
})

export const setCheckInEmail = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_EMAIL.REQUEST,
  payload,
})

export const setCheckInMobile = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_MOBILE.REQUEST,
  payload,
})

export const setCheckInCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_COUNTRY,
})

export const setCheckInCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_CALLING_CODE,
})

export const setCheckInOrder = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_ORDER.REQUEST,
  payload,
})

export const loadPrivacyPolicyDataSuccess = (payload: PrivacyPolicy, backgroundTask: boolean)
  : LoadPrivacyPolicySuccess => ({
    type: KIOSK.LOAD_PRIVACY_POLICY_DATA.SUCCESS,
    payload,
    backgroundTask
  })

export const addPrivateCustomerToQueueRequest = (queueId?: number, productId?: number)
  : AddPrivateCustomerToQueueRequest => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.REQUEST,
    productId,
    queueId,
})

export const addPrivateCustomerToQueueError = (payload: ErrorPayload) => ({
  type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE,
  payload
})

export const addPrivateCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddPrivateCustomerToQueueSuccess => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const addCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddCustomerToQueueSuccess => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const secretTap = () => ({
  type: APP.SECRET_TAP.REQUEST
})

export const addCustomerToQueueReset = (backgroundTask: boolean = false)
  : AddCustomerToQueueReset => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.RESET,
    backgroundTask
  })

export const checkIn = (): Action => ({ type: KIOSK.CHECK_IN.REQUEST })

export const goToInitialScreen = (): Action => ({ type: APP.GO_INITIAL_SCREEN.REQUEST })

export const loadPrivacyPolicyDataRequest = (): Action => ({ type: KIOSK.LOAD_PRIVACY_POLICY_DATA.REQUEST })

export const agreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_AGREE })

export const disagreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_DISAGREE })

export const addPrivateCustomerToQueueFail = (): Action => ({ type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE })

export interface GetKioskIdAndDomainFromUrlAction {
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
}

export const getKioskIdAndDomainFromUrl = (): GetKioskIdAndDomainFromUrlAction => ({
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
})

export interface SetKioskIdAction {
    type: SET_KIOSK_ID,
    payload: {
        id: string;
    }
}

export const setKioskId = (id: string): SetKioskIdAction => ({
    type: SET_KIOSK_ID,
    payload: {
        id
    }
})

export interface SetKioskSerialAction extends Action {
    type: SET_KIOSK_SERIAL,
    payload: {
        serial: string;
    }
}

export const setKioskSerial = (serial: string): SetKioskSerialAction => ({
    type: SET_KIOSK_SERIAL,
    payload: {
        serial
    }
})

export interface SetKioskUrlAction extends Action {
  type: SET_KIOSK_URL,
  payload: {
    url: string;
  }
}

export const setKioskUrl = (url: string): SetKioskUrlAction => ({
  type: SET_KIOSK_URL,
  payload: {
    url
  }
})

export type Actions =
  | Action
  | DatePayload
  | GetKioskDataSuccess
  | GetKioskSettingsSuccess
  | LanguagePayload
  | ProductPayload
  | QuestionAnswerPayload
  | RegisterKioskFieldUpdate
  | RegisterKioskSuccess
  | StringPayload
  | SetKioskUrlAction
  | SetKioskSerialAction
  | SetKioskIdAction
