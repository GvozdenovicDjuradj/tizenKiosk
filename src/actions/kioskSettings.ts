import { Action } from "redux"
import { KIOSK_SETTINGS_FORM } from "./types"
import {
  FieldUpdatePayload,
  KioskSettingsFieldUpdate
} from "../interfaces"

export const updateSettingsField = (payload: FieldUpdatePayload): KioskSettingsFieldUpdate => ({
  type: KIOSK_SETTINGS_FORM.FIELD_UPDATE.REQUEST,
  payload
})

export const checkNetwork = (): Action => ({
  type: KIOSK_SETTINGS_FORM.CHECK_NETWORK.REQUEST
})

export const validateKioskSettings = (): Action => ({
  type: KIOSK_SETTINGS_FORM.VALIDATE.REQUEST
})

export const saveKioskSettingsChanges = (): Action => ({
  type: KIOSK_SETTINGS_FORM.SUBMIT.REQUEST
})
