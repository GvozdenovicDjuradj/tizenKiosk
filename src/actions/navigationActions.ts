import { NavigationActions } from "react-navigation"
import { AppScreens } from "../interfaces"
import { Actions } from "../actions"

export const goToPrivacyPolicyScreen = (): Actions => (
  NavigationActions.navigate({
    routeName: AppScreens.PRIVACY_POLICY,
    key: AppScreens.PRIVACY_POLICY,
  })
)

export const goToCustomerDetailsScreen = (): Actions => (
  NavigationActions.navigate({
    key: AppScreens.CUSTOMER_DETAILS,
    routeName: AppScreens.CUSTOMER_DETAILS
  })
)
export const goToServiceSelectionScreen = (): Actions => (
  NavigationActions.navigate({
    key: AppScreens.SERVICE_SELECTION,
    routeName: AppScreens.SERVICE_SELECTION
  })
)

export const goToEventCheckInScreen = (): Actions => (
  NavigationActions.navigate({
    key: AppScreens.EVENT_CHECK_IN,
    routeName: AppScreens.EVENT_CHECK_IN,
  })
)

export const goToCheckInScreen = (): Actions => (
    NavigationActions.navigate({
      key: AppScreens.CHECK_IN,
      routeName: AppScreens.CHECK_IN
  })
)

export const goToQueueFullOrNAScreen = (): Actions => (
    NavigationActions.navigate({
      key: AppScreens.QUEUE_FULL_OR_NA,
      routeName: AppScreens.QUEUE_FULL_OR_NA
  })
)

export const goToUnderCapacityScreen = (): Actions =>
  NavigationActions.navigate({
    key: AppScreens.QUEUE_UNDER_OCCUPANCY,
    routeName: AppScreens.QUEUE_UNDER_OCCUPANCY,
  });

export const goToQuestionsScreen = (): Actions => (
    NavigationActions.navigate({
      key: `${AppScreens.QUESTIONS}_0`,
      routeName: AppScreens.QUESTIONS
  })
)

export const goToQueueConfirmationScreen = (): Actions => (
    NavigationActions.navigate({
      key: AppScreens.QUEUE_CONFIRMATION,
      routeName: AppScreens.QUEUE_CONFIRMATION
  })
)

export const goToCheckInConfirmationScreen = (): Actions => (
  NavigationActions.navigate({
    key: AppScreens.CHECK_IN_CONFIRMATION,
    routeName: AppScreens.CHECK_IN_CONFIRMATION
  })
)

export const goToEventCheckInConfirmationScreen = (): Actions => (
    NavigationActions.navigate({
      key: AppScreens.EVENT_CHECK_IN_CONFIRMATION,
      routeName: AppScreens.EVENT_CHECK_IN_CONFIRMATION
  })
)
