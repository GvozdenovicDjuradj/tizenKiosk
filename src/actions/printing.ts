import * as constants from "../constants/printing"
import { AnyAction } from "redux"
import { CustomerInQueue } from "../interfaces"

export interface SetPrinterCallback extends AnyAction {
    payload: {
        callback: (data: string) => void
    };
    type: constants.SET_PRINTER_CALLBACK;
}

export interface PrintTicket extends AnyAction {
    type: constants.PRINT_TICKET;
    payload: {
        customer: CustomerInQueue;
        isTest: boolean;
    };
}

export interface PrintTicketFulfilled extends AnyAction {
    type: constants.PRINT_TICKET_FULFILLED
}

export type PrintingAction = SetPrinterCallback | PrintTicket | PrintTicketFulfilled

export function setPrinterCallback(callback: (data: string) => void): PrintingAction {
    return {
        payload: {
            callback
        },
        type: constants.SET_PRINTER_CALLBACK
    }
}

export function printTicket(customer: CustomerInQueue, isTest = false): PrintTicket {
    return {
        type: constants.PRINT_TICKET,
        payload: {
            customer,
            isTest
        }
    }
}

export function printTicketFulfilled(): PrintTicketFulfilled {
    return {
        type: constants.PRINT_TICKET_FULFILLED
    }
}
