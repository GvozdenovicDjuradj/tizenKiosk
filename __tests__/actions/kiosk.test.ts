import {
    setAddCustomerToQueueError,
    addCustomerToQueueSuccess,
    addCustomerToQueueReset,
    kioskSettingsSuccess, getKioskIdAndDomainFromUrl, setKioskId, setKioskSerial, setKioskUrl
} from "../../src/actions/kiosk"
import { KIOSK } from "../../src/actions/types"
import { createCustomerInQueue, createSettings } from "../fixtures"

describe("The kiosk action creators", () => {
    describe("setAddCustomerToQueueError", () => {
        it("should return the correct action", () => {
            // Given
            const error = "Error message"

            // When
            const action = setAddCustomerToQueueError(error)

            // Then
            expect(action).not.toBeNull()
            expect(action).toHaveProperty("type", KIOSK.CUSTOMER.ADD_TO_QUEUE.FAILURE)
            expect(action).toHaveProperty("payload", error)
        })
    })

    describe("addCustomerToQueueSuccess", () => {
        it("should return the correct action", () => {
            // Given
            const customerInQueue = createCustomerInQueue()
            const payload = {
                customerInQueue
            }

            // When
            const action = addCustomerToQueueSuccess(customerInQueue)

            // Then
            expect(action).not.toBeNull()
            expect(action).toHaveProperty("type", KIOSK.CUSTOMER.ADD_TO_QUEUE.SUCCESS)
            expect(action).toHaveProperty("payload", payload)
        })
    })

    describe("addCustomerToQueueReset", () => {
        it("should return the correct action", () => {
            // When
            const action = addCustomerToQueueReset(true)

            // Then
            expect(action).not.toBeNull()
            expect(action).toHaveProperty("type", KIOSK.CUSTOMER.ADD_TO_QUEUE.RESET)
            expect(action).toHaveProperty("backgroundTask", true)
        })
    })

    describe("kioskSettingsSuccess", () => {
        it("should return the correct action", () => {
            // Given
            const kioskSettings = createSettings()

            // When
            const action = kioskSettingsSuccess(kioskSettings)

            // Then
            expect(action).toMatchSnapshot()
        })
    })

    describe(`getKioskIdAndDomainFromUrl`, () => {
        it("should return the correct action", () => {
            // When
            const action = getKioskIdAndDomainFromUrl()

            // Then
            expect(action).toMatchSnapshot()
        })
    })

    describe(`setKioskId`, () => {
        it("should return the correct action", () => {
            // When
            const action = setKioskId("test")

            // Then
            expect(action).toMatchSnapshot()
        })
    })

    describe(`setKioskSerial`, () => {
        it("should return the correct action", () => {
            // When
            const action = setKioskSerial("test")

            // Then
            expect(action).toMatchSnapshot()
        })
    })

    describe(`setKioskUrl`, () => {
        it("should return the correct action", () => {
            // When
            const action = setKioskUrl("test")

            // Then
            expect(action).toMatchSnapshot()
        })
    })
})
