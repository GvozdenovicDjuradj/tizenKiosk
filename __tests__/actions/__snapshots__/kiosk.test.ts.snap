// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The kiosk action creators getKioskIdAndDomainFromUrl should return the correct action 1`] = `
Object {
  "type": "GET_KIOSK_ID_AND_DOMAIN_FROM_URL",
}
`;

exports[`The kiosk action creators kioskSettingsSuccess should return the correct action 1`] = `
Object {
  "payload": Object {
    "active": true,
    "assigned": true,
    "bookingWidgetLink": "some link",
    "description": "Test description text",
    "eventSeriesLink": "",
    "id": 1,
    "identifier": "identifier",
    "isPushlinkEnabled": "no",
    "lastOnline": 2018-09-11T10:00:00.000Z,
    "lastSoftwareUpdate": 2018-09-11T10:00:00.000Z,
    "products": Array [],
    "qudiniVersion": "1.0.0",
    "serial": "test",
    "template": Object {
      "backgroundColor": "#fff",
      "backgroundImage": null,
      "backgroundImageIsEnabled": false,
      "buttonTextColor": "#fff",
      "confirmationScreenToShow": "CURRENT_STORE_WAIT_TIME",
      "customTextTranslations": Object {},
      "customerScreenEmail": "OPTIONAL",
      "customerScreenGroupSize": "OPTIONAL",
      "customerScreenIsRequestOrderNumber": false,
      "customerScreenIsShowMobileWarning": false,
      "customerScreenNameField": "OPTIONAL",
      "customerScreenRequestMobileNumberWhen": "NEVER",
      "customerScreenRequestMpnQueueLength": 0,
      "customerScreenRequestMpnWaitTime": 0,
      "customerScreenRequestOrderNumber": "OPTIONAL",
      "dateCreated": 2018-09-11T10:00:00.000Z,
      "dateEdited": null,
      "enableBookingCheckin": false,
      "enableBookingWidget": false,
      "enableEventCheckin": false,
      "enableOrderLookupConfig": false,
      "enableScreensaver": false,
      "enableWalkin": true,
      "font": "Arial",
      "headingTextColor": "#fff",
      "id": 0,
      "languageIsDisplayOther": false,
      "languagePosition": null,
      "languages": Object {
        "mainLanguage": Object {
          "countryCallingCode": "+44",
          "countryIsoCode": "GB",
          "countryName": "United Kingdom",
          "languageIsoCode": "en",
          "languageName": "en",
        },
        "otherLanguages": Array [],
        "translations": Object {},
      },
      "logo": null,
      "name": "Test template",
      "noAvailableImage": Object {},
      "noAvailableImageUrl": undefined,
      "orderLookupConfig": null,
      "screenFlow": "",
      "screenSaverEnableInSeconds": 10,
      "secondaryTextColor": "#dfdfdf",
      "serviceButtonColor": "#fff",
      "serviceScreenIsWithIcons": false,
      "serviceShowWaitTime": true,
      "showNothing": false,
      "useDefaultStyleTemplateLogo": true,
      "videoURL": "",
      "welcomeButtonColor": "#00ff00",
      "welcomeScreenIsRemove": false,
    },
    "ticket": Object {
      "font": "Arial",
      "id": 0,
      "merchant": Object {
        "id": 0,
        "name": "Test Merchant",
      },
      "name": "Test ticket",
      "showAtTop": "no",
      "ticketFor": "me",
    },
    "venue": Object {
      "defaultCountryCode": "GB",
      "defaultLanguageIsoCode": "US",
      "id": 1,
      "isBookingEnabled": true,
      "isWalkinEnabled": true,
      "merchant": Object {
        "featureSettings": Object {
          "hasQudiniBrand": true,
        },
        "id": 123,
      },
      "name": "Test Venue 1",
    },
  },
  "type": "KIOSK_SETTINGS_SUCCESS",
}
`;

exports[`The kiosk action creators setKioskId should return the correct action 1`] = `
Object {
  "payload": Object {
    "id": "test",
  },
  "type": "SET_KIOSK_ID",
}
`;

exports[`The kiosk action creators setKioskSerial should return the correct action 1`] = `
Object {
  "payload": Object {
    "serial": "test",
  },
  "type": "SET_KIOSK_SERIAL",
}
`;

exports[`The kiosk action creators setKioskUrl should return the correct action 1`] = `
Object {
  "payload": Object {
    "url": "test",
  },
  "type": "SET_KIOSK_URL",
}
`;
